src/ngx_http_lua_semaphore.c

```diff
@@ -557,8 +557,9 @@ ngx_http_lua_ffi_sema_gc(ngx_http_lua_sema_t *sem)
         return;
     }
 
-    if (!ngx_queue_empty(&sem->wait_queue)) {
-        ngx_log_error(NGX_LOG_CRIT, ngx_cycle->log, 0,
+    if (!ngx_terminate && !ngx_quit
+        && !ngx_queue_empty(&sem->wait_queue)) {
```

Comment:
The coding style for splitted if conditions is wrong. We should use the following form instead:

```C
if (A
    && B
    && C)
{
    ...
}
```